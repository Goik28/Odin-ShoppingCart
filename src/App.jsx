import { BrowserRouter, Routes, Route } from "react-router-dom";
import { Footer } from "./components/Footer";
import { Header } from "./components/Header";
import { Home } from "./components/Home";
import { Shop } from "./components/Shop";
import { products } from "./products";
import "./index.css";
import { useState } from "react";

export default function App() {
  const [shoppingCart, setShoppingCart] = useState([]);

  // add basename="/Odin-ShoppingCart/" to BrowserRouter for deployment on gh-pages.
  return (
    <BrowserRouter>
      <Header shoppingCart={shoppingCart} setShoppingCart={setShoppingCart} />
      <Routes>
        <Route path="/" element={<Home />}></Route>
        <Route
          path="/products"
          element={
            <Shop
              productsList={products}
              shoppingCart={shoppingCart}
              setShoppingCart={setShoppingCart}
            />
          }
        ></Route>
      </Routes>
      <Footer />
    </BrowserRouter>
  );
}
