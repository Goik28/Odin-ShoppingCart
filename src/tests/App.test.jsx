import { render } from "@testing-library/react";
import App from "../App";

describe("App integration", () => {
  it("Verify store header", () => {
    const wrapper = render(<App />);
    expect(wrapper.getByRole("navigation")).toBeInTheDocument();
  });

  it("Verify store footer", () => {
    const wrapper = render(<App />);
    expect(wrapper.getByText(/Created by/)).toBeInTheDocument();
  });

  it("Takes app snapshot", () => {
    const wrapper = render(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
