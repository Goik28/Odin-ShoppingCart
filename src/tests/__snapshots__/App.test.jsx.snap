// Vitest Snapshot v1

exports[`App integration > Takes app snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navHeader"
      >
        <header>
          <div
            class="storeName"
          >
            ANOTHER
          </div>
          <div
            class="subName"
          >
            fake store
          </div>
        </header>
        <nav>
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/products"
            title="Go to our products page"
          >
            Products
          </a>
        </nav>
        <div
          class="cart"
          title="Shopping cart"
        >
          <img
            alt="Shopping-Cart"
            class="cartIcon"
            src="./assets/shopping-cart.png"
          />
        </div>
      </div>
      <main
        class="home"
      >
        <h1>
          This is 
          <span>
            ANOTHER
          </span>
           fake store!
          <p>
            Actually this is just so the developer could practice his
             
            <span>
              REACT
            </span>
             skills. Read the footer dudes!
          </p>
        </h1>
        <div
          class="clipPath"
        />
        <h2>
          Here, we 
          <span>
            PRETEND
          </span>
           to sell fake stuff.
          <p
            class="checkItems"
          >
            <a
              href="/products"
              title="Go to our products page"
            >
              Check our items!
            </a>
          </p>
        </h2>
      </main>
      <footer>
        Created by 
        <a
          href="https://github.com/Goik28"
        >
          Goik28
        </a>
         for the Shopping Cart project on The Odin Project course
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="navHeader"
    >
      <header>
        <div
          class="storeName"
        >
          ANOTHER
        </div>
        <div
          class="subName"
        >
          fake store
        </div>
      </header>
      <nav>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/products"
          title="Go to our products page"
        >
          Products
        </a>
      </nav>
      <div
        class="cart"
        title="Shopping cart"
      >
        <img
          alt="Shopping-Cart"
          class="cartIcon"
          src="./assets/shopping-cart.png"
        />
      </div>
    </div>
    <main
      class="home"
    >
      <h1>
        This is 
        <span>
          ANOTHER
        </span>
         fake store!
        <p>
          Actually this is just so the developer could practice his
           
          <span>
            REACT
          </span>
           skills. Read the footer dudes!
        </p>
      </h1>
      <div
        class="clipPath"
      />
      <h2>
        Here, we 
        <span>
          PRETEND
        </span>
         to sell fake stuff.
        <p
          class="checkItems"
        >
          <a
            href="/products"
            title="Go to our products page"
          >
            Check our items!
          </a>
        </p>
      </h2>
    </main>
    <footer>
      Created by 
      <a
        href="https://github.com/Goik28"
      >
        Goik28
      </a>
       for the Shopping Cart project on The Odin Project course
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
